File: CS50_2022/pset6/sentimental-cash/cash.py
from cs50 import get_float


# Prompts the user for the change owed
# and ensures it is non-negative
while True:
    cents = get_float("Change owed: ")
    if cents > 0:
        break
# Convert to cents to avoid floating point imprecision
cents = cents * 100
# Calculates the number of quarters
quarters = cents // 25
cents = cents - (quarters * 25)
# Calculates the number of dimes
dimes = cents // 10
cents = cents - (dimes * 10)
# Calculates the number of nickels
nickels = cents // 5
cents = cents - (nickels * 5)
# Calculates the number of pennies
pennies = cents
# Coins is the sum of all the coin values
coins = quarters + dimes + nickels + pennies
# Prints the number of coins
print(int(coins))

File: CS50_2021/pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    int coins = 0;

    //Aks the user for the change owed and ensures the value is a non-negative integer
    float change;
    do
    {
        change = get_float("Change owed: ");
    }
    while (change < 0);

    //Converts our float to an int of cents to mitigate precision errors
    int cents = round (change * 100);

    while (cents >= 100)
    {
        cents -= 100;
        coins += 4;
    }
    while (cents >= 25)
    {
        cents -= 25;
        coins++;
    }

    while (cents >= 10)
    {
        cents -= 10;
        coins++;
    }
    while (cents >= 5)
    {
        cents -= 5;
        coins++;
    }
    while (cents >= 1)
    {
        cents -= 1;
        coins++;
    }
    printf("%i\n", coins);
}